src/logics/Logic.cc

```diff
@@ -1394,7 +1403,7 @@ void
 Logic::dumpFunction(ostream& dump_out, const TemplateFunction& tpl_fun)
 {
     const std::string& name = tpl_fun.getName();
-    auto quoted_name = protectName(name, tpl_fun.getRetSort(), tpl_fun.getArgs().size() == 0);
+    auto quoted_name = protectName(name, tpl_fun.getRetSort(), tpl_fun.getArgs().size() == 0, tpl_fun.isInterpreted());
```

Comment:
In which situations is `TemplateFunction` interpreted and in which uninterpreted?